Path are configurable and created automatically

MeinUsername opened the pull request:
- Config file can bet set via env. variable (komunumoconfig)
- Database path is configurable via database.path
- Database path will be created automatically together with the database file
- If both not set, the old default behavoir takes place

Commit:
- Config file can bet set via env. variable (komunumoconfig)
- Database path is configable via database.path
- Database path will be created automatically together with the database file
- If both not set, the old default behavoir takes place

McPringle commented on src/main/kotlin/ch/komunumo/server/business/configuration/control/ConfigurationService.kt:
```diff
@@ -17,14 +17,17 @@
-
+    private const val defaultConfigurationFilename = "komunumo.cfg"
+    private val logger = KotlinLogging.logger {}
+    private const val envVariableName = "komunumoconfig"
```
Thank you very much for your PR! I would prefer to have environment variable names in UPPERCASE. That's more common.

McPringle replied:
And I think `envVariableName` is not a good variable name. ;-)
How about `environmentConfigurationFilename`? That would match with `defaultConfigurationFilename`.

McPringle commented on src/main/kotlin/ch/komunumo/server/business/configuration/control/ConfigurationService.kt:
```diff
@@ -33,13 +36,42 @@ object ConfigurationService {
         val properties = Properties()
-        val file = Paths.get(System.getProperty("user.home"), ".komunumo", configurationFilename).toFile()
+        val filePath = System.getenv(envVariableName) ?: ""
+        val file: File = configFile(filePath)
```
I had to read these lines twice to understand, what they do. Maybe it is better to extract that into a private function `buildConfigurationFilename` or similar.

MeinUsername replied:
Alternative: I move the filePath into the configFile routine and rename configFile into buildConfigFile?

McPringle commented on src/main/kotlin/ch/komunumo/server/business/configuration/control/ConfigurationService.kt:
```diff
@@ -33,13 +36,42 @@ object ConfigurationService {
+
+
         if (file.exists()) {
+            ConfigurationService.logger.info { "Using configuration: " + file.toString() }
```
Btw, you don't need to prefix the logger call with the class name. The `logger` variable is an instance variable.

McPringle commented on src/main/kotlin/ch/komunumo/server/business/configuration/control/ConfigurationService.kt:
```diff
@@ -33,13 +36,42 @@ object ConfigurationService {
             FileInputStream(file).use { stream -> properties.load(stream) }
+        } else {
+            ConfigurationService.logger.info { "No configuration file found under " + file.toString() }
+            println("No configuration file found under " + file.toString())
```
Komunumo will not work without a configuration file. Maybe it is a good idea to stop Komunumo if there was no configuration file found. Maybe with throwing a `FileNotFoundException`.

Commit:
- Config file can bet set via env. variable (komunumoconfig)
- Database path is configable via database.path
- Database path will be created automatically together with the database file
- If both not set, the old default behavoir takes place
- Implemented renameings

MeinUsername commented:
I implemented most of your suggestions.

McPringle commented:
Thank you very much for your work!
